### PR TITLE
feat: add wasm fallback plugin

### LINKED ISSUE #1919: wasmFallbackPlugin
https://github.com/rolldown/vite/blob/ededbc26533fc3ff7069e00ab8990ead9fd3cf64/packages/vite/src/node/plugins/wasm.ts#L79-L96

### PR DESCRIPTION
**Pull request by ikkz**
resolves #1919

### CODE REVIEW

File: crates/rolldown_plugin_wasm_fallback/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+impl Plugin for WasmFallbackPlugin {
+  fn name(&self) -> Cow<'static, str> {
+    Cow::Borrowed("builtin:wasm-fallback-plugin")
+  }
+
+  async fn load(&self, _ctx: &SharedPluginContext, args: &HookLoadArgs<'_>) -> HookLoadReturn {
+    if std::path::Path::new(args.id)
+      .extension()
+      .map_or(false, |ext| ext.eq_ignore_ascii_case("wasm"))
```
Comment (IWANABETHATGUY) [inline crates/rolldown_plugin_wasm_fallback/src/lib.rs]: Use `args.id.ends_with` instead, construct a `Path` is slow

File: packages/rolldown/tests/fixture.test.ts [hunk 2]
```diff
@@ -19,11 +19,20 @@ function main() {
...
+            nodePath.join(import.meta.dirname, dirPath),
+            testConfig,
+          )
+        } catch (err) {
+          if (testConfig.catchError) {
+            await testConfig.catchError(err)
+          } else {
+            throw err
+          }
+        }
```
Comment (IWANABETHATGUY) [inline packages/rolldown/tests/fixture.test.ts]:
```suggestion
          const output = await compileFixture(
            nodePath.join(import.meta.dirname, dirPath),
            testConfig,
          ).catch((err) => {
            if (testConfig.catchError) {
              testConfig.catchError(err)
              return
            }
            throw err
          })

```

File: packages/rolldown/tests/fixture.test.ts [hunk 3]
```diff
@@ -22,8 +22,14 @@ function main() {
...
           testConfig,
-        )
-        if (testConfig.afterTest) {
+        ).catch(async (err) => {
+          if (testConfig.catchError) {
+            await testConfig.catchError(err)
+            return
+          }
+          throw err
+        })
```
Comment (ikkz) [inline packages/rolldown/tests/fixture.test.ts:31]: An `async` function is passed to catch because it allows to throw new errors in TestConfig#catchError

**Review by IWANABETHATGUY — APPROVED** [review]
Thanks

**Pull request merged**